payments/paytm/views.py

```diff
@@ -51,8 +53,16 @@ def response(request):
             data_dict[key] = request.POST[key]
         verify = Checksum.verify_checksum(data_dict, MERCHANT_KEY, data_dict['CHECKSUMHASH'])
         if verify:
-            PaytmHistory.objects.create(user=request.user, **data_dict)
-            return render(request,"response.html",{"paytm":data_dict})
+            for key in request.POST:
+                if key == "BANKTXNID" or key == "RESPCODE":
+                    if request.POST[key]:
+                        data_dict[key] = int(request.POST[key])
+                    else:
+                        data_dict[key] = 0
+                elif key == "TXNAMOUNT":
+                    data_dict[key] = float(request.POST[key])
+            PaytmHistory.objects.create(user=settings.USER, **data_dict)
```

Comment:
"user=settings.USER" by using this user will be fetched from settings , therefore if two users are doing payments at the same time  the object will be saved for the user who overright  the settings.USER (whor started payment most recently) as this variable is being set in payment view.

Reply:
Yes, i didnt thought about that scenerio... I was trying request.user but it was returning anonymous and if is make login required then page refreshes and i lose local variables from that instance that why i used User variable. If we create a list of  user and since lists are mutable between different classes so we can use that to store reference of the username.
Can you think of any better possible solution to it?

Reply:
I also had the same issue of 'anonymous user' as website is not able to access session variables  (due to something related to 'csrf') when directed from another website(paytm portal). So i redirected all of the POST data to another of my own website view and saved the data there. You can see that here (https://github.com/jaswal72/upes-fipi-jigyasa/blob/master/payments/views.py).
I have redirected data from 'response' view to 'recipt' view.